pkg/services/govmomi/vcenter/clone.go

```diff
@@ -229,27 +218,31 @@ func Clone(ctx *context.VMContext, bootstrapData []byte, format bootstrapv1.Form
 			return errors.Wrapf(err, "unable to get storageProfileID from name %s for %q", ctx.VSphereVM.Spec.StoragePolicyName, ctx)
 		}
 
-		kind := []string{"Datastore"}
-		m := view.NewManager(ctx.Session.Client.Client)
-
-		v, err := m.CreateContainerView(ctx, ctx.Session.Client.Client.ServiceContent.RootFolder, kind, true)
-		if err != nil {
-			return errors.Wrapf(err, "unable to create container view for Datastore for %q", ctx)
-		}
-
-		var content []types.ObjectContent
-		err = v.Retrieve(ctx, kind, []string{"name"}, &content)
-		_ = v.Destroy(ctx)
-		if err != nil {
-			return errors.Wrapf(err, "unable to retrieve container view for Datastore for %q", ctx)
-		}
-
 		var hubs []pbmTypes.PbmPlacementHub
-		for _, ds := range content {
+
+		// If there's a datastore request, it should be the only one requested from Storage Policy
+		if datastoreRef != nil {
 			hubs = append(hubs, pbmTypes.PbmPlacementHub{
-				HubType: ds.Obj.Type,
-				HubId:   ds.Obj.Value,
+				HubType: datastoreRef.Type,
+				HubId:   datastoreRef.Value,
 			})
+		} else {
+			// Otherwise we should get just the Datastores connected to our pool
+			cluster, err := pool.Owner(ctx)
+			if err != nil {
+				return errors.Wrapf(err, "failed to get owning cluster for storage policy datastore")
```

Comment:
```suggestion
				return errors.Wrapf(err, "failed to get owning cluster of resourcepool %q to calculate datastore based on storage policy", pool)
```
Maybe? (not sure if we can just print the pool)

Reply:
will double check on monday, I think we need to print its name (not sure pool has a stringer method)

Reply:
Yeap, this should work:
```
failed to get owning cluster of resourcepool "ResourcePool:resgroup-9 @ /DC-A/host/Cluster1/Resources"
```

Reply:
Perfect, thx!